skin/frontend/base/default/js/oyst/oneclick.js

```diff
@@ -11,7 +11,68 @@
 /**
  * OneClick Customs JS
  */
+"use strict";
 
+/**
+ * Main function to init 1-Click
+ *
+ * @param productId
+ * @param oneClickUrl
+ * @param isProductAddtocartFormValidate
+ */
+function oystOneClick(productId, oneClickUrl, isProductAddtocartFormValidate) {
+    window.__OYST__ = window.__OYST__ || {};
+    window.__OYST__.getOneClickURL = function (cb, opts) {
+        opts = opts || {};
+
+        ready(function () {
+            var qty = $$("input[name='qty']")[0].value ? $$("input[name='qty']")[0].value : 1;
+
+            var form = new FormData();
+            form.append("productRef", productId);
+            form.append("quantity", qty);
+            form.append("variationRef", getSimpleProductId());
+
+            var settings = {
+                async: true,
+                cacheControl: "no-cache",
+                data: form,
+                method: "POST",
+                mimeType: "multipart/form-data",
+                url: oneClickUrl
+            };
+
+            var xhr = new XMLHttpRequest();
+            xhr.open(settings.method, settings.url, settings.async);
+            xhr.setRequestHeader('cache-control', settings.cacheControl);
+            xhr.onload = function () {
+                if (200 === xhr.status) {
+                    var data = JSON.parse(xhr.responseText);
+                    var isErrorsInForm = null;
```

Comment:
why not `false` by default?

Reply:
null means they no form validation, true/false, is for form validation.

Reply:
This is ok for you?